### PR TITLE
Comic page

### LINKED ISSUE #22: Comic page
### Relevant resources:

- [resursive svelete components](https://svelte.dev/repl/347b37e18b5d4a65bbacfd097536db02?version=3.24.0)

to render the nested comments provided by the backend as IDs we need to recursively fill the children comments:

```ts
const fill_children = (comment: ComicCommentResponse, comments: ComicCommentResponse[], limit: number) => {
    if (limit <= 0) {
        comment.child_comments = [];
        return;
    }

    comment.child_comments = comment.child_comments?.map((child_id) => {
        const child_comment = get_comment_by_id(child_id, comments);

        if (child_comment) {
            console.log(child_comment);
            fill_children(child_comment, comments, limit - 1);

            return child_comment;
        }
    });
}

const get_comment_by_id = (id: string, comments: ComicCommentResponse[]): ComicCommentResponse | undefined => {
    return comments.find((comment) => {
        return comment.id == id;
    });
}
```

### Todos:
- [x] render nested comments
    - [x] fill comment children in object
    - [x] list in UI (https://svelte.dev/repl/347b37e18b5d4a65bbacfd097536db02?version=3.24.0)
    - [x] add comment box and a message for non-logged in users
- [x] add a way to create chapters in the comic page ( could be a + button or something )
- [x] list chapters (maybe in a grid)

### PR DESCRIPTION
**Pull request by BKSalman**
closes #22 

### Todos:
- [x] comment boxes
- [x] reply boxes collapsed by default
- [x]  a message for non-logged in users in the comments box section
- [ ] ~~a way to create chapters in the comic page ( could be a + button or something )~~
- [ ] ~~list chapters (maybe in a grid)~~

### CODE REVIEW

**Comment by mohad12211** [comment]
- [x] add a button to collapse the comment's children
- [x] show all comments for now (remove the depth limit condition)

**Comment by BKSalman** [comment]
>  show all comments for now (remove the depth limit condition)

what do you mean by this? I don't think we have a limit

**Comment by mohad12211** [comment]
> > show all comments for now (remove the depth limit condition)
> 
> what do you mean by this? I don't think we have a limit

https://github.com/BKSalman/musawarah/blob/f9f4f37435642849759ed6d6cee1d499158e452d/client/src/routes/%5Busername%5D/%5Bcomic_id%5D/%2Bpage.server.ts#L32
https://github.com/BKSalman/musawarah/blob/f9f4f37435642849759ed6d6cee1d499158e452d/client/src/routes/%5Busername%5D/%5Bcomic_id%5D/%2Bpage.server.ts#L43-L46

**Comment by BKSalman** [comment]
idk recursive stuff are scary, I would say keep the limit but make it seemingly high like 20 or something

**Comment by mohad12211** [comment]
map won't execute its body if the array is empty, so it will stop when you reach a comment with no children

**Comment by BKSalman** [comment]
hmm, well it doesn't matter that much so I will just comment the code inside the `if` and add a comment to the code

**Comment by BKSalman** [comment]
>  a way to create chapters in the comic page ( could be a + button or something )

this part is better done on a separate PR because it needs a new page and stuff

**Comment by BKSalman** [comment]
>  list chapters (maybe in a grid)

will also move this to a separate PR because it needs some extra work to make it decent

File: src/comics/comic_comments/routes.rs [hunk 1]
```diff
@@ -166,48 +159,69 @@ pub async fn create_comment(
...
-                    .execute(transaction)
-                    .await?;
+                if let Some(parent_comment_id) = payload.parent_comment_id {
+                    let count = comic_comments_mapping::table
+                        .filter(comic_comments_mapping::parent_comment_id.eq(parent_comment_id))
+                        .select(diesel::dsl::count(comic_comments_mapping::child_comment_id))
+                        .get_result::<i64>(transaction)
+                        .await?;
+
+                    tracing::debug!("count: {count}");
```
Comment (mohad12211) [inline src/comics/comic_comments/routes.rs]: is this only for debugging?

File: client/src/routes/[username]/[comic_id]/+page.svelte [hunk 2]
```diff
@@ -1,14 +1,50 @@
...
+    import { goto } from '$app/navigation';
+    import type { ComicCommentResponse } from 'bindings/ComicCommentResponse';
+    import { faMessage } from '@fortawesome/free-solid-svg-icons';
+    import Fa from 'svelte-fa';
 
     export let data: PageServerData;
 
-    const { comic, comments } = data;
-</script>
+    let open = false;
```
Comment (mohad12211) [inline client/src/routes/[username]/[comic_id]/+page.svelte]:
```suggestion
```
unused variable

File: client/src/routes/[username]/[comic_id]/+page.svelte [hunk 3]
```diff
@@ -1,14 +1,50 @@
 <script lang="ts">
-    import { page } from '$app/stores';
     import type { PageServerData } from './$types';
     import ChapterThumbnail from '$lib/components/ChapterThumbnail.svelte';
     import Comment from '$lib/components/Comment.svelte';
+    import { currentUser } from '../../stores';
+    import { goto } from '$app/navigation';
+    import type { ComicCommentResponse } from 'bindings/ComicCommentResponse';
+    import { faMessage } from '@fortawesome/free-solid-svg-icons';
+    import Fa from 'svelte-fa';
```
Comment (mohad12211) [inline client/src/routes/[username]/[comic_id]/+page.svelte]:
```suggestion
```
unused variables

**Review by mohad12211 — CHANGES_REQUESTED** [review]
just unused code resulting from copy pasting I assume

File: src/comics/comic_comments/routes.rs (hunk 1 above)
Comment (BKSalman) [reply src/comics/comic_comments/routes.rs]: I added it a while ago to try and limit the nesting, but it can be removed it's useless

Comment (mohad12211) [reply src/comics/comic_comments/routes.rs]: yeah we should remove it for now

Comment (BKSalman) [inline src/comics/comic_comments/routes.rs]:
```suggestion
```

**Comment by BKSalman** [comment]
LGTM?

**Review by mohad12211 — APPROVED** [review]
LGTM LGTM !!

**Comment by BKSalman** [comment]
LGTM!!!

**Pull request merged**